client/app/scripts/superdesk-legal-archive/legal_archive.js

```diff
@@ -149,14 +164,25 @@
 
         $scope.clear = function () {
             legal.criteria = $scope.criteria = {};
-            $scope.items = legal.default_items;
+            $scope.search();
         };
 
         $scope.$watch(function getSearchParams() {
             return _.omit($location.search(), '_id');
         }, refresh, true);
 
+        $scope.setview = function(view) {
+            $scope.view = view || 'mgrid';
+            viewUpdate['archive:view'].view = view || 'mgrid';
+            preferencesService.update(viewUpdate, 'archive:view');
+        };
+
         $scope.search();
+
+        preferencesService.get('archive:view').then(function(result) {
+            var savedView = result.view;
+            $scope.view = (!!savedView && savedView !== 'undefined') ? savedView : 'mgrid';
```

Comment:
Was this perhaps meant to be `typeof savedView !== 'undefined'` or are we indeed comparing `savedView` to a literal string `'undefined'`?